resource-service/common/git.go

```diff
@@ -135,12 +140,42 @@ func (g Git) storeDefaultBranchConfig(gitContext common_models.GitContext, err e
 	return nil
 }
 
+func retrieveDefaultBranchFromEnv(env envconfig.EnvConfig) string {
```

Comment:
this could also be a member function of `EnvConfig` -> just to reduce the verbosity of the git.go file a bit